### ISSUE TITLE
Are these lines written backwards in this GRIB file, and can cfgrib correct this?

### ISSUE THREAD

**Issue by blaylockbk** [ecmwf/cfgrib#240]
I download a GRIB2 file from NOAA's National Blend of Models from [NOMADS](https://nomads.ncep.noaa.gov/pub/data/nccf/com/blend/prod/) and plotted some fields. It looks like the data is written backwards in some rows! Can cfgrib fix this and read the lines correctly? Here is an example of 2-m temperature

```python
import xarray as xr
import cfgrib
import urllib

# to reproduce, you'll need to change the URL to a recent path from 
# https://nomads.ncep.noaa.gov/pub/data/nccf/com/blend/prod/
url = 'https://nomads.ncep.noaa.gov/pub/data/nccf/com/blend/prod/blend.20210728/00/core/blend.t00z.core.f001.co.grib2'
file = url.split('/')[-1]
urllib.request.urlretrieve(url, file)

z = xr.open_dataset(
    file,
    engine='cfgrib', 
    backend_kwargs={'filter_by_keys': {'shortName': '2t', 'typeOfLevel': 'heightAboveGround'}}
)

z.t2m.plot(x='longitude', y='latitude')
z.t2m.plot()
```
![image](https://user-images.githubusercontent.com/6249613/127377252-c18758cf-96cb-4990-b92c-dcc4a9b12677.png)
![image](https://user-images.githubusercontent.com/6249613/127377272-93a885df-f133-4d18-9a34-cbcb90400898.png)

**Comment by blaylockbk** [comment]
Same issue as #276

**Comment by shahramn** [comment]
This will be fixed in ecCodes v2.25.0
